std/typecons.d

```diff
@@ -5509,19 +5509,19 @@ if (!is(T == class) && !(is(T == interface)))
             import core.exception : onOutOfMemoryError;
             import core.stdc.string : memcpy, memset;
 
-            _store = cast(Impl*) pureMalloc(Impl.sizeof);
+            _store = (() @trusted => cast(Impl*) pureMalloc(Impl.sizeof))();
             if (_store is null)
                 onOutOfMemoryError();
             static if (hasIndirections!T)
-                pureGcAddRange(&_store._payload, T.sizeof);
+                (() @trusted => pureGcAddRange(&_store._payload, T.sizeof))();
 
             // Can't use std.algorithm.move(source, _store._payload)
             // here because it requires the target to be initialized.
             // Might be worth to add this as `moveEmplace`
 
             // Can avoid destructing result.
             static if (hasElaborateAssign!T || !isAssignable!T)
-                memcpy(&_store._payload, &source, T.sizeof);
+                (() @trusted => memcpy(&_store._payload, &source, T.sizeof))();
```

Comment:
`@trusted` for the memcpy. Okay because this cannot cannot cause memory corruption barring unsafe previous operations. By specification it is not `safe` for any structure to contain a pointer to itself, so the change in address is allowed; _store is non-null; and _store has the correct size.